src/openms_gui/source/VISUAL/SpectraTreeTab.cpp

```diff
@@ -182,12 +182,18 @@ namespace OpenMS
     }
   }
 
+  void SpectraTreeTab::updateIndexFromCurrentLayer()
+  {
+    spectra_treewidget_->setTreePosition(layer_->getCurrentSpectrumIndex());
+  }
+
   void SpectraTreeTab::itemSelectionChange_(QTreeWidgetItem* current, QTreeWidgetItem* previous)
   {
     /*	test for previous == 0 is important - without it,
         the wrong spectrum will be selected after finishing
         the execution of a TOPP tool on the whole data */
-    if (current == nullptr || previous == nullptr)
+    // WUT????
+    if (current == nullptr)
```

Comment:
has this been verified?
The test for `previous` was there for a reason?!

Reply:
Not really, but this cannot be the solution to whatever problem this is trying to solve.
With this, you cannot ever set a selection if nothing was selected previously.

Reply:
The problem was, if i remember correctly, that TOPPView crashed.
Hence the extra check.
So now it crashes again, given the right circumstances, but we can set a selection?
Seems like this needs some investigation...

Reply:
Yes true, I think whatever made it crash should probably be fixed instead of restricting other functions just to be safe.

@timosachsenberg can you quickly check? You have it built already.

Reply:
sure what exactly?

Reply:
Where the comment says

Reply:
Sorry on phone

Reply:
Didn't crash now

Reply:
Still? Or worse? 😆

Reply:
hehe no still :) the usual strange stuff in TOPPView

Reply:
Ok yes, I remember still having issues as well.